applications/GeoMechanicsApplication/custom_utilities/solving_strategy_factory.hpp

```diff
@@ -0,0 +1,96 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "solving_strategies/strategies/solving_strategy.h"
+#include "spaces/ublas_space.h"
+#include "linear_solvers/linear_solver.h"
+#include "solving_strategies/convergencecriterias/mixed_generic_criteria.h"
+#include "solving_strategy_factory.hpp"
+#include "factories/standard_linear_solver_factory.h"
+#include "solving_strategies/strategies/residualbased_linear_strategy.h"
+#include "custom_strategies/strategies/geo_mechanics_newton_raphson_strategy.hpp"
+#include "scheme_factory.hpp"
+#include "convergence_criteria_factory.hpp"
+#include "builder_and_solver_factory.hpp"
+
+namespace Kratos
+{
+
+template<class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class SolvingStrategyFactory
+{
+public:
+    [[nodiscard]] static std::unique_ptr<SolvingStrategy<TSparseSpace, TDenseSpace>> Create(Parameters& rSolverSettings, ModelPart& rModelPart)
```

Comment:
If I'm not mistaken, I think we can make the first parameter a reference to const:
```suggestion
    [[nodiscard]] static std::unique_ptr<SolvingStrategy<TSparseSpace, TDenseSpace>> Create(const Parameters& rSolverSettings, ModelPart& rModelPart)
```

Reply:
Done